### PR TITLE
Encrypt/decrypt stored change blocks on IPFS.

### PR DESCRIPTION
**Pull request by robotoer**
I think we can skip encrypting/decrypting sync messages and rely upon libp2p pipe encryption (otherwise we'd be double encoding). Data that goes over the pubsub topic does still need to be encrypted/decrypted (coming in next change).

### CODE REVIEW

File: packages/collabswarm/src/auth-subtlecrypto.ts [hunk 1]
```diff
@@ -71,11 +71,10 @@ export class SubtleCrypto
   _encryptionAlgorithmParams(nonce?: Uint8Array): AesGcmParams {
     switch (this._encryptionAlgorithmName) {
       case "AES-GCM":
-        let iv_value = crypto.getRandomValues(new Uint8Array(this._nonceBits));
-        if (nonce) iv_value = nonce;
+        const iv = nonce ? nonce : crypto.getRandomValues(new Uint8Array(this.nonceBits));
```
Comment (lifekaizen) [inline packages/collabswarm/src/auth-subtlecrypto.ts]: Oh yeah this is better, skip the function if not needed.

File: packages/collabswarm/src/auth-subtlecrypto.ts [hunk 2]
```diff
@@ -71,11 +71,10 @@ export class SubtleCrypto
   _encryptionAlgorithmParams(nonce?: Uint8Array): AesGcmParams {
     switch (this._encryptionAlgorithmName) {
```
Comment (lifekaizen) [inline packages/collabswarm/src/auth-subtlecrypto.ts:72]: Do we just want to remove `_encryptionAlgorithmName` since it's not used anymore?

Comment (robotoer) [reply packages/collabswarm/src/auth-subtlecrypto.ts:72]: Isn't this called still in both `decrypt` and `encrypt`?

File: packages/collabswarm-yjs/bin/collabswarm-yjs-d.ts [hunk 3]
```diff
@@ -14,14 +14,14 @@ const keychain = new YjsKeychainProvider();
...
   .generateKey(
     {
-      name: "AES-GCM",
-      length: 256,
+      name: "ECDSA",
+      namedCurve: "P-384",
     },
     true,
-    ["encrypt", "decrypt"]
+    ["sign", "verify"]
```
Comment (lifekaizen) [inline packages/collabswarm-yjs/bin/collabswarm-yjs-d.ts]: Maybe we can discuss more offline - curious about background, also why bin is in repo

**Review by lifekaizen — APPROVED** [review]

File: packages/collabswarm/src/auth-subtlecrypto.ts (hunk 2 above)
Comment (lifekaizen) [reply packages/collabswarm/src/auth-subtlecrypto.ts:72]: They call this `_encryptionAlgorithmParams` function, not the `_encryptionAlgorithmName` directly

**Review by lifekaizen — APPROVED** [review]

**Pull request merged**